### PR TITLE
Langfuse: Add comprehensive error handling and tests

### PR DESCRIPTION
**Pull request by vprashrex**
- Added try-catch blocks to prevent Langfuse failures from blocking main flow
- Ensure tracer is always initialized (disabled if no credentials)
- Added graceful degradation for all tracer methods
- Created comprehensive test suite covering all failure scenarios
- Tests verify OpenAI responses work regardless of Langfuse state

Fixes issue where Langfuse connection failures could block API responses.

## Summary

Target issue is #510 
Explain the **motivation** for making this change. What existing problem does the pull request solve?

## Summary by CodeRabbit

* **Bug Fixes**
  * Observability/tracing is now defensive: initialization/runtime failures are caught, tracing becomes a safe no-op when unavailable, errors no longer affect core flows, session/context resumption is preserved, and richer metadata is recorded on success.
* **Tests**
  * Added comprehensive unit and integration tests covering init, failure modes, no-op semantics, error handling, full tracing lifecycle, end-to-end response processing, and a fixture to bypass global DB seeding during these tests.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

File: backend/app/core/langfuse/langfuse.py [hunk 1]
```diff
@@ -29,28 +29,45 @@ def __init__(
...
-            logger.info(
-                f"[LangfuseTracer] Langfuse tracing enabled | session_id={self.session_id}"
-            )
-        else:
-            self.langfuse = Langfuse(enabled=False)
-            logger.warning(
-                "[LangfuseTracer] Langfuse tracing disabled due to missing credentials"
-            )
+                logger.info(
+                    f"[LangfuseTracer] Langfuse tracing enabled | session_id={self.session_id}"
```
Comment (AkhileshNegi) [inline backend/app/core/langfuse/langfuse.py]: this is somehow still populating in the logs even though the credentials are revoked

**Review by AkhileshNegi — APPROVED** [review]

File: backend/app/tests/core/test_langfuse/test_langfuse_tracer.py [hunk 2]
```diff
@@ -0,0 +1,849 @@
...
+from openai import OpenAIError
+
+from app.core.langfuse.langfuse import LangfuseTracer
+from app.models import Assistant, ResponsesAPIRequest
+from app.services.response.response import generate_response, process_response
+
+
+# =============================================================================
+# Fixtures
+# =============================================================================
```
Comment (AkhileshNegi) [inline backend/app/tests/core/test_langfuse/test_langfuse_tracer.py]: lots of unnecessary formatting

**Review by AkhileshNegi — APPROVED** [review]

File: backend/app/tests/core/test_langfuse/test_langfuse_tracer.py [hunk 3]
```diff
@@ -0,0 +1,849 @@
...
+Tests all scenarios to ensure:
+1. OpenAI responses are ALWAYS generated regardless of Langfuse state
+2. Langfuse failures never block the main execution flow
+3. All edge cases are handled gracefully
+
+Scenarios covered:
+- No credentials (None)
+- Incomplete credentials (missing keys)
+- Langfuse initialization exception
+- fetch_traces failure (non-critical)
```
Comment (Prajna1999) [inline backend/app/tests/core/test_langfuse/test_langfuse_tracer.py]: nitpick: isn't fetch_traces failure is critical? Like am I missing some context here

File: backend/app/core/langfuse/langfuse.py [hunk 4]
```diff
@@ -59,56 +76,87 @@ def start_trace(
...
+                )
+        except Exception as e:
+            logger.error(f"[LangfuseTracer] Failed to log error: {e}")
 
     def flush(self):
-        self.langfuse.flush()
+        if not self.langfuse.enabled:
+            return
+        try:
+            self.langfuse.flush()
```
Comment (Prajna1999) [inline backend/app/core/langfuse/langfuse.py]: Is `try_catch` really needed for one line method invocation? May be keep the one liner `langfuse.flush()` wherever it needs to be invoked without this extra abstraction.

File: backend/app/core/langfuse/langfuse.py [hunk 5]
```diff
@@ -138,6 +186,13 @@ def wrapper(
                 logger.info("[Langfuse] No credentials - skipping observability")
                 return func(completion_config, query, **kwargs)
 
+            def safe_langfuse_op(op: Callable, *args, **kwargs):
+                try:
+                    return op(*args, **kwargs)
```
Comment (Prajna1999) [inline backend/app/core/langfuse/langfuse.py]: ditto as above. Does really a oneliner require its own `try_catch`

File: backend/app/core/langfuse/langfuse.py [hunk 6]
```diff
@@ -59,56 +76,87 @@ def start_trace(
...
+        try:
             self.generation.end(output=output, usage=usage, model=model)
+        except Exception as e:
+            logger.error(f"[LangfuseTracer] Failed to end generation: {e}")
 
     def update_trace(self, tags: list[str], output: Dict[str, Any]):
-        if self.trace:
-            self.trace.update(tags=tags, output=output)
+        if not self.langfuse.enabled or not self.trace:
+            return
```
Comment (Prajna1999) [inline backend/app/core/langfuse/langfuse.py:123]: ditto as above

**Review by Prajna1999 — COMMENTED** [review]
The overall sudo_code is
`if langfuse_fail:
       continue
` 
I think there are far simpler ways to implement this logic instead of having individual `try_catch` blocks for oneliner method invocations.

File: backend/app/core/langfuse/langfuse.py [hunk 7]
```diff
@@ -144,30 +176,56 @@ def wrapper(
...
+                    f"[observe_llm_execution] Failed to initialize client: {e}"
+                )
                 return func(completion_config, query, **kwargs)
 
-            trace = langfuse.trace(
+            failed = False
+
+            def langfuse_call(fn, *args, **kwargs):
+                """Execute Langfuse operation safely. First failure disables further calls."""
+                nonlocal failed
```
Comment (Prajna1999) [inline backend/app/core/langfuse/langfuse.py:192]: what does `nonlocal` do?

**Review by Prajna1999 — APPROVED** [review]
general comment: the code has become a bit hard to read. Let's see if we can make the implementation cleaner and easy to follow. If this is the best we can do then LGTM
